bigframes/core/log_adapter.py

```diff
@@ -0,0 +1,38 @@
+import functools
+
+from loguru import logger
+
+_log_file_path = None
+_logger = logger
+_api_methods = []
+
+
+def class_logger(decorated_cls):
+    """Decorator that adds logging functionality to each method of the class."""
+    for attr_name, attr_value in decorated_cls.__dict__.items():
+        if callable(attr_value):
+            setattr(decorated_cls, attr_name, method_logger(attr_value))
+    return decorated_cls
```

Comment:
This would be where we'd override the initializer, I think to create an `_api_methods` attribute, potentially combining the `_api_methods` from objects passed into the constructor.